src/psyclone/psyir/nodes/node.py

```diff
@@ -1149,6 +1149,40 @@ def scope(self):
             "Unable to find the scope of node '{0}' as none of its ancestors "
             "are Container or Schedule nodes.".format(self))
 
+    def replace_with(self, node):
+        '''Replaces self with the supplied node in the PSyIR tree.
+
+        :param node: the node that will replace self in the PSyIR \
+            tree.
+        :type node: subclass of :py:class:`psyclone.psyir.nodes.node`
+
+        :raises TypeError: if the argument 'node' is not a Node.
+        :raises SymbolError: if this node does not have a parent.
+        :raises SymbolError: if the argument 'node' has a parent.
+
+        '''
+        if not isinstance(node, Node):
+            raise TypeError(
+                "The argument node in method replace_with in the Node class "
+                "should be a Node but found '{0}'."
+                "".format(type(node).__name__))
+        if not self.parent:
+            raise GenerationError(
+                "This node should have a parent if its replace_with method "
+                "is called.")
+        if node.parent is not None:
+            raise GenerationError(
+                "The parent of argument node in method replace_with in the "
+                "Node class should be None but found '{0}'."
+                "".format(type(node.parent).__name__))
+
+        parent = self.parent
+        position = self.position
+        node.parent = parent
+        self.parent.children.remove(self)
```

Comment:
I think this remove operation is dangerous when removing a children and the index position is significant (e.g. children 0 of IfBlock which should be an expression - a Schedule can not be displaced into it). Could you add a test for this?
Anyway, this two lines can be converted to `self.parent.children[position] = node` to avoid the issue.

Reply:
That's much simpler, thanks. If I understand correctly, there is no additional test required now that I've changed to your proposed much simpler solution.

Reply:
I would still prefer the test in case somebody changed this implementation the issue should be evident, and at the moment because replace_with is still not used across the code it wouldn't fail anywhere else.

Just replacing the "Schedule<-Statement, Statement, ..." construct in test_replace_with() by a construct where order is important would already do.

Reply:
OK, I've added an additional test.